karamel-core/src/main/java/se/kth/karamel/backend/launcher/occi/OcciLauncher.java

```diff
@@ -0,0 +1,330 @@
+import cz.cesnet.cloud.occi.infrastructure.NetworkInterface;
+import java.io.File;
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import se.kth.karamel.backend.launcher.Launcher;
+import se.kth.karamel.backend.running.model.ClusterRuntime;
+import se.kth.karamel.common.clusterdef.json.JsonCluster;
+import se.kth.karamel.common.exception.InvalidOcciCredentialsException;
+import se.kth.karamel.common.exception.KaramelException;
+import se.kth.karamel.common.util.Confs;
+import se.kth.karamel.common.util.OcciCredentials;
+import se.kth.karamel.common.util.SshKeyPair;
+import java.util.List;
+import java.util.Set;
+import org.apache.log4j.Logger;
+import se.kth.karamel.backend.converter.UserClusterDataExtractor;
+import se.kth.karamel.backend.running.model.GroupRuntime;
+import se.kth.karamel.backend.running.model.MachineRuntime;
+import se.kth.karamel.common.clusterdef.Occi;
+import se.kth.karamel.common.clusterdef.Provider;
+import se.kth.karamel.common.clusterdef.json.JsonGroup;
+import se.kth.karamel.common.util.Settings;
+
+
+/**
+ * Created by Mamut on 2016-1-19.
+ * Class required to launch machines in OCCI compatible clouds
+ */
+public final class OcciLauncher extends Launcher{
+    
+  private static final int DEFAULT_SSH_PORT = 22;  
+  private static final Logger logger = Logger.getLogger(OcciLauncher.class);
+  public final OcciContext context;
+  private static boolean TESTING = true;  
+  private final SshKeyPair sshKeyPair;
+  
+  /**
+   *
+   * @param occiContext
+   * @param sshKeyPair
+   */
+  public OcciLauncher(OcciContext occiContext, SshKeyPair sshKeyPair) {
+    this.context = occiContext;
+    this.sshKeyPair = sshKeyPair;
+  }
+
+  /**
+   * Dummy validate function
+   * @param occiCredentials
+   * @return
+   * @throws InvalidOcciCredentialsException
+   */
+  public static OcciContext validateCredentials(OcciCredentials occiCredentials)
+          throws InvalidOcciCredentialsException {
+    OcciContext context = new OcciContext(occiCredentials);
+      
+    //basic path and file existence check
+    File f = new File(context.getOcciCredentials().getUserCertificatePath());    
+    if((!f.exists() || f.isDirectory())){
+      logger.info(String.format("Cert does not exist in provided path" + 
+              context.getOcciCredentials().getUserCertificatePath()));
+      throw new InvalidOcciCredentialsException("Cert does not exist in provided path" + 
+              context.getOcciCredentials().getUserCertificatePath());  
+    }
+      
+    f = new File(context.getOcciCredentials().getSystemCertDir());
+    if ((!f.exists() || !f.isDirectory())) {
+      logger.info(String.format("Cert folder does not exist in provided path" + 
+              context.getOcciCredentials().getSystemCertDir()));
+      throw new InvalidOcciCredentialsException("Cert folder does not exist in provided path" + 
```

Comment:
I think the exception message needs a space character before the appended path

Maybe you should explicitly spell "certificate" instead of "cert"